File: javascript/01-arrays/max.js
function maxnum(arr){
    let currentMax = arr[0];
    for(let i = 1; i < arr.length; i++){
        if(arr[i]> currentMax){
            currentMax = arr[i]
        }
    }
    return currentMax

}

function maxnumInRange(arr, start, end){
    
    if (start > end){
        return -1
    }

    let currentMax = arr[0];
    for(let  i = start; i < end + 1; i++){
        if(arr[i] > currentMax){
            currentMax = arr[i]
        }

    }
    return currentMax
}


let arr = [ 3,5,8,7,12]
console.log(maxnum(arr))
console.log(maxnumInRange(arr, 0 , 3))

File: python/02-linearSearch/max.py
def max(arr):
    current_max = arr[0]
    for index in range(1, len(arr)):
        if arr[index] > current_max:
            current_max = arr[index]
    return current_max


def max_range(arr, start, end):
    if start > end:
        return -1

    current_max = arr[start]
    for index in range(start, end +1):
        if arr[index]> current_max:
            current_max = arr[index]
    
    return current_max

arr = (1, 3, 2, 9, 18)
print(max(arr=arr))
print(max_range(arr=arr, start=0, end=2))
